### ISSUE TITLE
Rule Change: `array-bracket-newline` should respect single element arrays on "multiline"-mode

### ISSUE THREAD

**Issue by 4lph4-Ph4un** [eslint/eslint#15719]
### What rule do you want to change?

array-bracket-newline

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

Other

### Example code

```js
const aVeryLongArrayNameForAnArrayWeMightAppendLater = [
  aVeryLongVariableNameIncomingBooyaah
];
```

### What does the rule currently do for this code?

The rule enforces me to change the the array into a oneliner like this:
```javascript
const aVeryLongArrayNameForAnArrayWeMightAppendLater = [aVeryLongVariableNameIncomingBooyaah];
```
This is, however, suboptimal in cases where dealing with very long variable names/elements or when we wish to declare an array that is to be added with new elements, preferably not causing changes for the original line (minimizing diffs).

How this is to be achieved? Really no idea at this point! Posting this as a starter for a conversation about this.

### What will the rule do after it's changed?

To not warn/error on single element arrays that are intended to be the start of a beautiful multielement, multiline array!

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by aladdin-add** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.
